subprojects/analysis-kotlin-api/src/main/kotlin/org/jetbrains/kotlin/analysis/kotlin/internal/SampleProvider.kt

```diff
@@ -0,0 +1,23 @@
+package org.jetbrains.kotlin.analysis.kotlin.internal
+
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.InternalDokkaApi
+
+@InternalDokkaApi
+interface SampleProviderFactory {
+    fun build(): SampleProvider
+}
```

Comment:
It's a bit unclear why `SampleProviderFactory` is needed at this moment - will there be scenarios in the near future where it's possible to have multiple providers, or what was the idea behind it?

Reply:
`SampleProvider` is a short-life instance (but it takes memory enough during its existence) that the transformer only needs.

Reply:
oh, makes sense. The `Factory` pattern made me think that there could be multiple implementations of `SampleProvider`. I have an idea about how to make the API friendlier here without it, but no time to discuss it, unfortunately :( I guess I'll try to remember about it later

Could you please document that it's used for creating short-lived instances due to memory consumption?